homeassistant/helpers/config_validation.py

```diff
@@ -147,6 +147,31 @@ def entity_ids(value: Union[str, Sequence]) -> Sequence[str]:
     return [entity_id(ent_id) for ent_id in value]
 
 
+class EntitiesDomain():
```

Comment:
I don't think you need a class. You can make a factory function (nested) where the outer function takes the domain and the inner the entities as parameters. This pattern is already used here in config validation.